apps/src/applab/commands.js

```diff
@@ -434,42 +436,34 @@ applabCommands.getDirection = function(opts) {
 applabCommands.dot = function(opts) {
   apiValidateTypeAndRange(opts, 'dot', 'radius', opts.radius, 'number', 0.0001);
   var ctx = applabTurtle.getTurtleContext();
+  var isPenDown = Applab.turtle.isPenDown;
   if (ctx && opts.radius > 0) {
-    ctx.beginPath();
-    if (Applab.turtle.penUpColor) {
-      // If the pen is up and the color has been changed, use that color:
-      ctx.strokeStyle = Applab.turtle.penUpColor;
+    if (!isPenDown) {
+      // If Applab.turtle pen is up, temporarily set Applab.turtle pen down to draw dot
+      Applab.turtle.isPenDown = true;
```

Comment:
Where is this value used in the code below, before it's set to `false` again a little [lower](https://github.com/code-dot-org/code-dot-org/pull/49726/files#diff-ddc4837b291d272c88f8e4cb34bda64ba45763193c98d213ed4b5032ba311e01R454)?